plugin/table/table.go

```diff
@@ -103,58 +103,96 @@ func (t *Plugin) Ping() osquery.ExtensionStatus {
+// https://github.com/osquery/osquery/blob/master/osquery/core/sql/column.h#L37
+type ColumnOptions uint8
+
+// From https://github.com/osquery/osquery/blob/master/osquery/core/sql/column.h#L37
+const (
+	ColumnOptionDefault    ColumnOptions = 0
+	ColumnOptionIndex                    = 1
+	ColumnOptionRequired                 = 2
+	ColumnOptionAdditional               = 4
+	ColumnOptionOptimized                = 8
+	ColumnOptionHidden                   = 16
+)
+
 // ColumnDefinition defines the relevant information for a column in a table
 // plugin. Both values are mandatory. Prefer using the *Column helpers to
 // create ColumnDefinition structs.
 type ColumnDefinition struct {
-	Name string
-	Type ColumnType
+	Name        string
+	Type        ColumnType
+	Options     ColumnOptions // bitmask. See https://github.com/osquery/osquery/blob/master/osquery/core/sql/column.h#L37
+	Description string
 }
 
-// TextColumn is a helper for defining columns containing strings.
-func TextColumn(name string) ColumnDefinition {
-	return ColumnDefinition{
+func newColumn(name string, ctype ColumnType, opts ...ColumnOpt) ColumnDefinition {
+	cd := ColumnDefinition{
 		Name: name,
-		Type: ColumnTypeText,
+		Type: ctype,
 	}
+
+	for _, opt := range opts {
+		opt(&cd)
+	}
+
+	return cd
+
+}
+
+// TextColumn is a helper for defining columns containing strings.
+func TextColumn(name string, opts ...ColumnOpt) ColumnDefinition {
+	return newColumn(name, ColumnTypeText, opts...)
 }
 
 // IntegerColumn is a helper for defining columns containing integers.
-func IntegerColumn(name string) ColumnDefinition {
-	return ColumnDefinition{
-		Name: name,
-		Type: ColumnTypeInteger,
-	}
+func IntegerColumn(name string, opts ...ColumnOpt) ColumnDefinition {
+	return newColumn(name, ColumnTypeInteger, opts...)
 }
 
 // BigIntColumn is a helper for defining columns containing big integers.
-func BigIntColumn(name string) ColumnDefinition {
-	return ColumnDefinition{
-		Name: name,
-		Type: ColumnTypeBigInt,
-	}
+func BigIntColumn(name string, opts ...ColumnOpt) ColumnDefinition {
+	return newColumn(name, ColumnTypeBigInt, opts...)
 }
 
 // DoubleColumn is a helper for defining columns containing floating point
 // values.
-func DoubleColumn(name string) ColumnDefinition {
-	return ColumnDefinition{
-		Name: name,
-		Type: ColumnTypeDouble,
+func DoubleColumn(name string, opts ...ColumnOpt) ColumnDefinition {
+	return newColumn(name, ColumnTypeDouble, opts...)
+}
+
+func IndexColumn() ColumnOpt      { return SetColumnOption(ColumnOptionIndex) }
+func RequiredColumn() ColumnOpt   { return SetColumnOption(ColumnOptionRequired) }
+func AdditionalColumn() ColumnOpt { return SetColumnOption(ColumnOptionAdditional) }
+func OptimizedColumn() ColumnOpt  { return SetColumnOption(ColumnOptionOptimized) }
+func HiddenColumn() ColumnOpt     { return SetColumnOption(ColumnOptionHidden) }
+
+func SetColumnOption(flag ColumnOptions) ColumnOpt {
```

Comment:
Any consideration to making some fo these types(and helper functions) private? I would be in favor if not exposing `setColumnOption`. It can always be exposed later if needed, but it's hard to hide. 

I'd also consider if `ColumnDefinition.Options` can be private., only allowing options to be set by passing in a function parameter.

Reply:
I was trying to decide what the right model is. Whether it's cleaner to expose `IndexColumn()` or `SetColumnOption(...ColumnOptions)`

Having slept on it, I think it's probably cleaner to expose only `IndexColumn`

I sorta wish we could go back and replace `TextColumn` with `NewColumn(foo, TextColumn())` but that feels like too much of a change

Reply:
I made more stuff internal. I think that's a good call. But also made a lot of functional arguments